test/support/barriers.h

```diff
@@ -0,0 +1,42 @@
+// -*- C++ -*-
+//===---------------------------------------------------------------------===//
+//
+// Copyright (C) 2023 Intel Corporation
+//
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef _ONEDPL_BARRIERS_H
+#define _ONEDPL_BARRIERS_H
+
+
+#include <thread>
+#include <mutex>
+#include <condition_variable>
+
+class Barrier {
+public:
+    explicit Barrier(std::size_t count) : threshold(count), count(count), generation(0) {}
```

Comment:
I think we will have incorrect work here if `count` == 0 or `threshold` == 0.
I believe `assert` may help to check this.